src/WCSimConstructCylinder.cc

```diff
@@ -288,49 +286,66 @@ G4LogicalVolume* WCSimDetectorConstruction::ConstructCylinder()
+    G4Tubs *solidCaveCapsTyvekTop = new G4Tubs("CaveCapsTyvekTop",
                                             0,
                                             WCRadius,
                                             .5 * (WCODTyvekSheetThickness),
                                             0. * deg,
                                             360. * deg);
 
-    G4LogicalVolume *logicCaveCapsTyvek =
-	  new G4LogicalVolume(solidCaveCapsTyvek,
-						  G4Material::GetMaterial("Tyvek"),
-						  "CaveCapTyvek",
-						  0, 0, 0);
+    G4Tubs *solidCaveCapsTyvekBottom = new G4Tubs("CaveCapsTyvekBottom",
+                                            0,
+                                            WCRadius,
+                                            .5 * (WCODTyvekSheetThickness),
+                                            0. * deg,
+                                            360. * deg);
 
-    //G4LogicalSkinSurface *TyvekCaveTopSurface =
-	new G4LogicalSkinSurface("TyvekCaveTopSurface", logicCaveCapsTyvek, OpWaterTySurface);
+    G4LogicalVolume *logicCaveCapsTyvekTop =
+			new G4LogicalVolume(solidCaveCapsTyvekTop,
+								G4Material::GetMaterial("Tyvek"),
+								"CaveCapTyvek",
+								0, 0, 0);
 
-    G4VisAttributes *CapsCaveTyvekVisAtt = new G4VisAttributes(yellow);
-    CapsCaveTyvekVisAtt->SetForceWireframe(true);
-    logicCaveCapsTyvek->SetVisAttributes(CapsCaveTyvekVisAtt);
+    G4LogicalVolume *logicCaveCapsTyvekBottom =
+			new G4LogicalVolume(solidCaveCapsTyvekBottom,
+								G4Material::GetMaterial("Tyvek"),
+								"CaveCapTyvek",
+								0, 0, 0);
+
+    G4LogicalSkinSurface *TyvekCaveTopSurface = new G4LogicalSkinSurface("TyvekCaveTopSurface", logicCaveCapsTyvekTop, OpWaterTySurfaceOutWallTop);
+    G4LogicalSkinSurface *TyvekCaveBottomSurface = new G4LogicalSkinSurface("TyvekCaveBottomSurface", logicCaveCapsTyvekBottom, OpWaterTySurfaceOutWallBottom);
+
+    G4VisAttributes *CapsCaveTyvekVisAttTop = new G4VisAttributes(yellow);
+    CapsCaveTyvekVisAttTop->SetForceWireframe(true);
+    logicCaveCapsTyvekTop->SetVisAttributes(CapsCaveTyvekVisAttTop);
+
+    G4VisAttributes *CapsCaveTyvekVisAttBottom = new G4VisAttributes(yellow);
+    CapsCaveTyvekVisAttBottom->SetForceWireframe(true);
+    logicCaveCapsTyvekBottom->SetVisAttributes(CapsCaveTyvekVisAttBottom);
     //logicCaveCapsTyvek->SetVisAttributes(G4VisAttributes::Invisible); //amb79
 
     G4ThreeVector CaveTyvekPosition(0., 0., WCLength / 2);
 
-    //G4VPhysicalVolume *physiTopCaveTyvek =
-	new G4PVPlacement(0,
-					  CaveTyvekPosition,
-					  logicCaveCapsTyvek,
-					  "CaveTopTyvek",
-					  logicWCBarrel,
-					  false,
-					  checkOverlaps);
+    G4VPhysicalVolume *physiTopCaveTyvek =
+        new G4PVPlacement(0,
+                          CaveTyvekPosition,
+						  logicCaveCapsTyvekTop,
+                          "CaveTopTyvek",
+						  logicWCBarrel,
+                          false,
+                          0);
```

Comment:
Please revert. As it was, we can easily turn overlap checks on via compilation options. Without, we can't
```suggestion
                          checkOverlaps);
```